org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/component/Messagebox.java

```diff
@@ -294,6 +321,12 @@ public int show(String message, String title, int buttons, String icon, WEditor
 
 		if ((buttons & NO) != 0)
 			btnNo.setVisible(true);
+		
+		if ((buttons & DELETE) != 0)
+			btnDelete.setVisible(true);
+
+		if ((buttons & CANCEL_TEXT) != 0)
+			btnCancelText.setVisible(true);
```

Comment:
I'm skeptical that this is necessary. For me, "to proceed with delete" is still a question to the user and a standard yes, no, cancel should work just fine too.

Reply:
We think using the new "DELETE" and "CANCEL_TEXT" buttons would be more user friendly and would follow today's standards. This UI would be more clear to Users - e.g. they could know the purpose of the dialog (to delete records) without even reading the question.

Reply:
According to some comments from @norbertbede the intention of this is not to be used just on delete cases, this can be also used in case other dangerous operations in the system, for example before running a process that is dangerous.

Because of that I agree with Heng Sin a normal Yes/No dialog would suffice, and indeed that makes it consistent with the actual delete dialog that will appear in all the other windows that don't have a "Delete Confirmation Logic" configured.

I think this also suggests that the method FDialog.askForInputDeleteConfirmation must be changed to askForInputTextConfirmation (or something like that).


About text buttons, I think it's better to make the UI consistent, if you want text buttons, then it is better to implement a way in FDialog to manage text buttons everywhere (maybe a Tenant SysConfig?) - again here the user will see some dialogs with icon buttons deleting in some windows, and text buttons in other windows, IMO that's bad user experience.